assets/js/lib/filters/mealsMapFilters.js

```diff
@@ -0,0 +1,22 @@
+class MealsMapFilters {
+    constructor(onChangeCallback) {
+      this.parentMapFilters = new MapFilters({
+        filterNames: [
+          "shouldHaveFree",
+        ],
+        filtersElemSelector: ".meals-filters input",
+        onChangeCallback: onChangeCallback
+      });
+  
+      this.anyApplied = this.parentMapFilters.anyApplied;
+    }
+  
+    shouldShowLocation(location) {
+    //   if (this.parentMapFilters.state.shouldHaveFree && location.hasFree) {
```

Comment:
@Rakshith-2905 Would you be able to implement this logic?

Filters are not working because `cowinMap` is checking if any of the filters (`MealsMapFilters`, `HospitalFilters` or `OxygenSupplyFilters`) are returning `true` for the `shouldShowLocation`. 

This filter here is making every marker to be displayed because it's returning `true`. You need to implement the logic.

Can you implement the logic looking at other filters?

Reply:
modified this on this commit 

https://github.com/Cowin-team/cowin_map/commit/5d37f54f02e85c7b3b55971850177c7b2a73cec7